walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -21,9 +21,28 @@
 import java.util.Iterator;
 
 public final class FindMeetingQuery {
+  /* If we can accomodate optional attendees, do so. Otherwise if there are required attendees,
+  ignore optional attendees.*/
+  public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
+    Collection<String> totalAttendees = new ArrayList<String>();
+    totalAttendees.addAll(request.getAttendees());
+    totalAttendees.addAll(request.getOptionalAttendees());
+
+    MeetingRequest requestWithOptionals = new MeetingRequest(totalAttendees, 
+        request.getDuration());
+    Collection<TimeRange> optResults = query2(events, requestWithOptionals);
+    
+    if ((optResults.isEmpty()) && (!request.getAttendees().isEmpty())) {
+      return query2(events, request);
+    }
+    else {
+      return optResults;
+    }
+  }
+
   /* Identify conflicting events, then iterating through the day, skipping conflicts, and 
   identifying time ranges longer than the duration of the requested meeting. */
-  public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
+  public Collection<TimeRange> query2(Collection<Event> events, MeetingRequest request) {
```

Comment:
Adding a number like 2 to the end of a method doesn't give the reader any extra information (unless `2` has some meaning here that I'm not aware of). I think you should just keep this function's name `query`.

Reply:
The tests (most of which we did not write) depend on our overall 'main' function being named 'query.' Therefore, I can name my internal query function anything except 'query' (unless I want two functions named query). I will try queryInternal, that at least gives more info than query2?